workflow/controller/scope.go

```diff
@@ -38,23 +35,16 @@ func (s *wfScope) addArtifactToScope(key string, artifact wfv1.Artifact) {
 
 // resolveVar resolves a parameter or artifact
 func (s *wfScope) resolveVar(v string) (interface{}, error) {
-	v = strings.TrimPrefix(v, "{{")
-	v = strings.TrimSuffix(v, "}}")
-	parts := strings.Split(v, ".")
-	prefix := parts[0]
-	switch prefix {
-	case "steps", "tasks", "workflow":
-		val, ok := s.scope[v]
-		if ok {
-			return val, nil
-		}
-	case "inputs":
-		art := s.tmpl.Inputs.GetArtifactByName(parts[2])
-		if art != nil {
-			return *art, nil
+	m := make(map[string]interface{})
+	for k, v := range s.scope {
+		m[k] = v
+	}
+	if s.tmpl != nil {
+		for _, a := range s.tmpl.Inputs.Artifacts {
+			m["inputs."+a.Name] = a // special case for artifacts
```

Comment:
I think the key should be `inputs.artifacts '+a.Name

Reply:
correct. i misunderstnand the old value

Reply:
you are correct